### PR TITLE
GCSBucketIOMock compares generated files with files on disk

### PR DESCRIPTION
**Pull request by intrepiditee**

### CODE REVIEW

File: import-automation/executor/test/test_integration.py [hunk 1]
```diff
@@ -75,17 +76,22 @@ class DashboardAPIMock(mock.MagicMock):
...
 
     def upload_file(self, src, dest):
         self.data[dest] = src
         with open(src) as file:
             logging.warning(f'Generated {src}: {file.readline()}')
+        assert filecmp.cmp(src, os.path.join(CWD, 'test', 'data', os.path.basename(src)), shallow=False)
 
     def update_version(self, version):
-      logging.warning(f'Version: {version}')
+        logging.warning(f'Version: {version}')
```
Comment (pradh) [inline import-automation/executor/test/test_integration.py:94]: mock time and validate version?

File: import-automation/executor/test/test_integration.py [hunk 2]
```diff
@@ -75,17 +76,22 @@ class DashboardAPIMock(mock.MagicMock):
     #     return run
 
 
+CWD = os.getcwd()
+
+
 class GCSBucketIOMock:
     def __init__(self, path_prefix='', bucket_name='', bucket=None, client=None):
         self.data = {}
+        self.prefix = path_prefix
```
Comment (pradh) [inline import-automation/executor/test/test_integration.py:85]: Where is self.prefix used?

Validate its content?

File: import-automation/executor/test/test_integration.py [hunk 3]
```diff
@@ -75,17 +76,22 @@ class DashboardAPIMock(mock.MagicMock):
...
 
     def upload_file(self, src, dest):
         self.data[dest] = src
         with open(src) as file:
             logging.warning(f'Generated {src}: {file.readline()}')
+        assert filecmp.cmp(src, os.path.join(CWD, 'test', 'data', os.path.basename(src)), shallow=False)
```
Comment (pradh) [inline import-automation/executor/test/test_integration.py:91]: Can we trim the file to first N lines or something and validate so we don't need largeish files in the repo?

File: import-automation/executor/test/test_integration.py (hunk 1 above)
Comment (intrepiditee) [reply import-automation/executor/test/test_integration.py:94]: aded

File: import-automation/executor/test/test_integration.py (hunk 2 above)
Comment (intrepiditee) [reply import-automation/executor/test/test_integration.py:85]: Not used here. I forgot to remove this line

File: import-automation/executor/test/test_integration.py (hunk 3 above)
Comment (intrepiditee) [reply import-automation/executor/test/test_integration.py:91]: fixed

File: import-automation/executor/test/test_integration.py [hunk 4]
```diff
@@ -75,6 +76,17 @@ class DashboardAPIMock(mock.MagicMock):
...
 
 
+CWD = os.getcwd()
+
+
+def _compare_lines(path1, path2, num_lines):
+    with open(path1) as file1, open(path2) as file2:
+        for i in range(num_lines):
+            if file1.readline() != file2.readline():
+                return False
```
Comment (pradh) [inline import-automation/executor/test/test_integration.py:86]: Dump the file contents so its easy for user to debug?

Comment (intrepiditee) [reply import-automation/executor/test/test_integration.py:90]: added

**Pull request merged**